sherpa/astro/tests/test_astro_data2.py

```diff
@@ -6158,3 +6169,412 @@ def test_rmf_get_dep_complex():
+               ("TFORM1", "1I"),
+               ("TLMIN1", str(minchan)),
+               ("TLMAX1", str(minchan + nrows - 1)),
+               ("TTYPE2", "E_MIN"),
+               ("TFORM2", "1E"),
+               ("TUNIT2", "keV"),
+               ("TTYPE3", "E_MAX"),
+               ("TFORM3", "1E"),
+               ("TUNIT3", "keV"),
+              ]
+
+    with open(filename, mode="w+b") as fh:
+        add_header(fh, primary)
+        add_header(fh, matrix)
+
+        # The heap is organized as
+        # - f_chan[0], n_chan[0], matrix[0]
+        # - f_chan[1][0], f_chan[1][1], n_chan[1][0], n_chan[1][1],
+        #   matrix[1][0], matrix[1][1]
+        # - f_chan[2], n_chan[2], matrix[2]
+        # - f_chan[3], n_chan[3], matrix[3]
+        #
+        size = 0
+        heap_size = 0
+        heap_int = struct.calcsize(">h")
+        for elo, ehi, ngrp in zip(energ_lo, energ_hi, n_grps):
+            if ngrp == 1:
+                fh.write(struct.pack(matrix_fmt, elo, ehi, 1,
+                                     1, heap_size,
+                                     1, heap_size + heap_int,
+                                     1, heap_size + 2 * heap_int))
+                heap_size += matrix_heap_size1
+
+            elif ngrp == 2:
+                fh.write(struct.pack(matrix_fmt, elo, ehi, 2,
+                                     2, heap_size,
+                                     2, heap_size + 2 * heap_int,
+                                     2, heap_size + 4 * heap_int))
+                heap_size += matrix_heap_size2
+
+            else:
+                raise RuntimeError("internal error")
+
+            size += matrix_size
+
+        # Add in the heap.
+        #
+        for fc, nc, mz in zip(f_chans, n_chans, matrixs):
+            for v in fc:
+                fh.write(struct.pack(">h", v))
+            for v in nc:
+                fh.write(struct.pack(">h", v))
+            for v in mz:
+                fh.write(struct.pack(">f", v))
+
+        size += matrix_heap_size
+        add_spacer(fh, size)
+        add_header(fh, ebounds)
+
+        size = 0
+        for elo, ehi, chan in zip(energ_lo, energ_hi, chans):
+            fh.write(struct.pack(ebounds_fmt, chan, elo, ehi))
+            size += ebounds_size
+
+        add_spacer(fh, size)
+
+    return chans, energ_lo, energ_hi
+
+
+@requires_fits
+@pytest.mark.xfail(is_crates_io, reason="tests known to fail with crates")
+@pytest.mark.parametrize("offset", [0, 1, 5])
+def test_problem_pha(tmp_path, offset):
+    """Check we can read in the 'problem' FITS file"""
+
+    from sherpa.astro.io import read_pha
+
+    outpath = tmp_path / 'tmp.pha'
+    chans, counts, quals = make_problem_pha(outpath, offset)
+
+    pha = read_pha(str(outpath))
+    assert isinstance(pha, DataPHA)
+
+    assert pha.channel[0] == offset
+
+    assert pha.channel == pytest.approx(chans)
+    assert pha.counts == pytest.approx(counts)
+    # convert bools to nitegers to please pytest 8.3.4
```

Comment:
nitegers -> integers. Not that it matters, but just so you know I actually read every line...